### PR TITLE
feat: migrate redis to instrumentation #246

### LINKED ISSUE #246: Convert redis plugin to instrumentation
- Change folder name from `opentelemetry-plugin-*` to `opentelemetry-instrumentation-*`
- Change package name from `@opentelemetry/plugin-*` to `@opentelemetry/instrumentation-*`
- Change implementation to use [InstrumentationBase](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts#L30)

### PR DESCRIPTION
**Pull request by vmarchaud**
Nothing changed a part migrating to instrumentation base class

Fixes #246

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-redis/package.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  "devDependencies": {
+    "@opentelemetry/context-async-hooks": "0.18.0",
+    "@opentelemetry/node": "0.18.0",
+    "@opentelemetry/tracing": "0.18.0",
+    "@opentelemetry/test-utils": "0.14.0",
+    "@types/mocha": "7.0.2",
+    "@types/node": "14.0.27",
+    "@types/redis": "2.8.25",
+    "@types/shimmer": "1.0.1",
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-redis/package.json:53]: I guess @types/shimmer is no longer needed

File: plugins/node/opentelemetry-instrumentation-redis/package.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "rimraf": "3.0.2",
+    "ts-mocha": "8.0.0",
+    "ts-node": "9.0.0",
+    "tslint-consistent-codestyle": "1.16.0",
+    "tslint-microsoft-contrib": "6.2.0",
+    "typescript": "4.1.3"
+  },
+  "dependencies": {
+    "@opentelemetry/api": "^0.18.0",
+    "@opentelemetry/core": "^0.18.0",
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-redis/package.json:69]: Is core still needed?

File: plugins/node/opentelemetry-instrumentation-redis/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Apache 2.0 - See [LICENSE][license-url] for more information.
+
+[gitter-image]: https://badges.gitter.im/open-telemetry/opentelemetry-js.svg
+[gitter-url]: https://gitter.im/open-telemetry/opentelemetry-node?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+[license-url]: https://github.com/open-telemetry/opentelemetry-js/blob/main/LICENSE
+[license-image]: https://img.shields.io/badge/license-Apache_2.0-green.svg?style=flat
+[dependencies-image]: https://david-dm.org/open-telemetry/opentelemetry-js/status.svg?path=packages/opentelemetry-instrumentation-redis
+[dependencies-url]: https://david-dm.org/open-telemetry/opentelemetry-js?path=packages%2Fopentelemetry-instrumentation-redis
+[devDependencies-image]: https://david-dm.org/open-telemetry/opentelemetry-js/dev-status.svg?path=packages/opentelemetry-instrumentation-redis
+[devDependencies-url]: https://david-dm.org/open-telemetry/opentelemetry-js?path=packages%2Fopentelemetry-instrumentation-redis&type=dev
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-redis/README.md:67]: some links should be updated to point to opentelemetry-js-contrib instead opentelemetry-js

**Review by Flarna — APPROVED** [review]

**Review by dyladan — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-redis/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+### Supported Versions
+ - `>=2.6.0`
+
+## Usage
+
+OpenTelemetry Redis Instrumentation allows the user to automatically collect trace data and export them to the backend of choice, to give observability to distributed systems when working with [redis](https://www.npmjs.com/package/redis).
+
+To load a specific instrumentation (**redis** in this case), specify it in the registerInstrumentations's configuration
```
Comment (obecny) [inline plugins/node/opentelemetry-instrumentation-redis/README.md:25]:
```suggestion
To load a specific instrumentation (**redis** in this case), specify it in the registerInstrumentations' configuration
```

**Review by obecny — APPROVED** [review]

**Pull request merged**